packages/grid/_modules_/grid/hooks/features/rows/useGridRowsHydration.ts

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+import { GridEvents } from '../../../models/events/gridEvents';
+import { useGridState } from '../../utils/useGridState';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { GridRowsHydrationApi } from '../../../models/api/gridRowsHydrationApi';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { getCurrentPageRows } from '../../utils/useCurrentPageRows';
+import { GridPreProcessingGroup } from '../../core/preProcessing';
+import { GridEventListener } from '../../../models/events/gridEventListener';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridRowId } from '../../../models/gridRows';
+import {
+  GridRowHeightParams,
+  GridRowHeightReturnValue,
+} from '../../../models/params/gridRowParams';
+
+/**
+ * @requires useGridRowGroupsPreProcessing (method)
+ * @requires useGridPageSize (method)
+ * @requires useGridPage (method)
+ */
+export const useGridRowsHydration = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'getRowHeight' | 'pagination' | 'paginationMode'>,
+): void => {
+  const [gridState, setGridState, forceUpdate] = useGridState(apiRef);
+  const rowsHeightCollection = React.useRef(new Map<GridRowId, number>());
+
+  const hydrateRowsMeta = React.useCallback(
```

Comment:
I tried that approach but the thing is that it's the same logic as the columns. It's a different data structure now, so instead of having just a static `rowHeight` we extract the row height by checking the positions of the different rows.